### PR TITLE
PLATFORM-1232 force a FromSession user in Piggyback sessions

### PR DESCRIPTION
**Pull request by ghost**
When an access token is provided, the user object for the request
is created based on that. It makes Piggyback useless as Piggyback
relies on session data accessed later during the execution.

We can distinguish a Piggyback session from a regular one by
a simple check and override the user object created from an
access token with one created from session data.

@wladekb @macbre

### CODE REVIEW

File: includes/context/RequestContext.php [hunk 1]
```diff
@@ -206,8 +206,9 @@ public function getUser() {
 		if ( $this->user === null && $wgUserForceAnon ) {
 			$this->user = new User();
 		}
+		// replace the FromToken user with a FromSession user when in a Piggyback session
+		if ( $this->user === null || PBLoginForm::isPiggyback() ) {
```
Comment (wladekb) [inline includes/context/RequestContext.php]: is the PBLoginForm class always available when this function can be called?
AFAIR it's our extension that can be at least theoretically disabled, I'd suggest putting
a check if that class exists

Comment (michalroszka) [inline includes/context/RequestContext.php]: Based on the same input from @grunny and my own second thoughts (great timing), I have decided to make the core change more general and move the Piggyback-specific logic to Piggyback. :-)

**Comment by Grunny** [comment]
LGTM :shipit:

**Comment by Grunny** [comment]
LGTM... again. :)

**Pull request merged**